### ISSUE TITLE
Error: YAML file error: MOTR_CONFIG section is missing

### ISSUE THREAD

**Issue by daniarherikurniawan** [Seagate/cortx-hare#1661]
Hi guys, I am following the readme to install Hare. When I run `sudo m0crate -S /tmp/m0crate-io.yaml`, I got this error:

> error: YAML file error: MOTR_CONFIG section is missing
> error: Failed to parse Z��
> Unable to parse workload:-22

I attached my /tmp/m0crate-io.yaml
[m0crate-io.txt](https://github.com/Seagate/cortx-hare/files/6689450/m0crate-io.txt)

Let me know if you guys know how to fix the yaml file. I believe there is no problem with m0crate itself.

Thanks

**Comment by vaibhavparatwar** [comment]
@knekrasov any idea about this ?

**Comment by pavankrishnat** [comment]
Hi @daniarherikurniawan,

Looks like there is some problem with m0crate file. I tried same at my end, it ran successfully without any error. Here is my output attached.
`
[root@ssc-vm-5571 ~]# hctl bootstrap --mkfs ~/cluster.yaml
2021-06-23 03:56:34: Generating cluster configuration... OK
2021-06-23 03:56:37: Stopping Consul on this node... OK
2021-06-23 03:56:53: Stopping Consul on other nodes... OK
2021-06-23 03:57:10: Starting Consul server on this node......... OK
2021-06-23 03:57:17: Importing configuration into the KV store... OK
2021-06-23 03:57:17: Starting Consul on other nodes.... OK
2021-06-23 03:57:22: Updating Consul configuraton from the KV store... OK
2021-06-23 03:57:24: Waiting for the RC Leader to get elected.... OK
2021-06-23 03:57:25: Starting Motr (phase1, mkfs)... OK
2021-06-23 03:58:25: Starting Motr (phase1, m0d)... OK
2021-06-23 03:58:42: Starting Motr (phase2, mkfs)... OK
2021-06-23 04:00:41: Starting Motr (phase2, m0d)... OK
2021-06-23 04:00:46: Starting S3 servers (phase3)... OK
2021-06-23 04:00:47: Checking health of services... OK
[root@ssc-vm-5571 ~]# /opt/seagate/cortx/hare/libexec/m0crate-io-conf >/tmp/m0crate-io.yaml
Usage: grep [OPTION]... PATTERN [FILE]...
Try 'grep --help' for more information.
Usage: grep [OPTION]... PATTERN [FILE]...
Try 'grep --help' for more information.
Usage: grep [OPTION]... PATTERN [FILE]...
Try 'grep --help' for more information.
Usage: grep [OPTION]... PATTERN [FILE]...
Try 'grep --help' for more information.
Usage: grep [OPTION]... PATTERN [FILE]...
Try 'grep --help' for more information.
Usage: grep [OPTION]... PATTERN [FILE]...
Try 'grep --help' for more information.
Usage: grep [OPTION]... PATTERN [FILE]...
Try 'grep --help' for more information.
Usage: grep [OPTION]... PATTERN [FILE]...
Try 'grep --help' for more information.
Usage: grep [OPTION]... PATTERN [FILE]...
Try 'grep --help' for more information.
[root@ssc-vm-5571 ~]# cat /tmp/m0crate-io.yaml
CrateConfig_Sections: [MOTR_CONFIG, WORKLOAD_SPEC]

MOTR_CONFIG:
   MOTR_LOCAL_ADDR: 192.168.84.113@tcp:12345:4:1
   MOTR_HA_ADDR:    192.168.84.113@tcp:12345:1:1
   PROF: <0x7000000000000001:0xc6>  # Profile fid
   LAYOUT_ID: 9                      # Defines the UNIT_SIZE (9: 1MB)
   IS_OOSTORE: 1                     # Is oostore-mode?
   IS_READ_VERIFY: 0                 # Enable read-verify?
   TM_RECV_QUEUE_MIN_LEN: 16  # Minimum length of the receive queue
   MAX_RPC_MSG_SIZE: 65536    # Maximum rpc message size
   PROCESS_FID: <0x7200000000000001:0x26>
   IDX_SERVICE_ID: 1

LOG_LEVEL: 4  # err(0), warn(1), info(2), trace(3), debug(4)

WORKLOAD_SPEC:                # Workload specification section
   WORKLOAD:                  # First Workload
      WORKLOAD_TYPE: 1        # Index(0), IO(1)
      WORKLOAD_SEED: tstamp   # SEED to the random number generator
      OPCODE: 3               # Operation(s) to test: 2-WRITE, 3-WRITE+READ
      IOSIZE: 10m      # Total Size of IO to perform per object
      BLOCK_SIZE: 2m          # In N+K conf set to (N * UNIT_SIZE) for max perf
      BLOCKS_PER_OP: 1        # Number of blocks per motr operation
      MAX_NR_OPS: 1           # Max concurrent operations per thread
      NR_OBJS: 10             # Number of objects to create by each thread
      NR_THREADS: 4           # Number of threads to run in this workload
      RAND_IO: 1              # Random (1) or sequential (0) IO?
      MODE: 1                 # Synchronous=0, Asynchronous=1
      THREAD_OPS: 0           # All threads write to the same object?
      NR_ROUNDS: 1            # Number of times this workload is run
      EXEC_TIME: unlimited    # Execution time (secs or "unlimited")
      SOURCE_FILE: /tmp/128M  # Source data file
[root@ssc-vm-5571 ~]# dd if=/dev/urandom of=/tmp/128M bs=1M count=128
128+0 records in
128+0 records out
134217728 bytes (134 MB) copied, 1.10077 s, 122 MB/s
[root@ssc-vm-5571 ~]# sudo m0crate -S /tmp/m0crate-io.yaml
info: starting workload 0
info: workload type          io/2
info: random seed:           0
info: number of threads:     4
trace: oid 0004d0435f109d57:309b5744155a23c2
trace: oid 00d0e2ed2c7313b4:591604f610014680
trace: oid 00b8797d33bd637f:158366da12fee890
trace: oid 007977d418efcc40:2fd8f8441f6a73da
trace: oid 008e0f987af21622:32498c4e66aca7f7
trace: oid 000a871e693f5685:7c363a1427f6ee03
trace: oid 00a71e9175aad8f3:239d3d722438672f
trace: oid 003aca5929c36462:79280d1d3c2d29fb
trace: oid 00d401b92a9d52b3:51874dbd173f9a9c
trace: oid 001066673cc447f6:2740e11c6fa4e4a6
trace: oid 0081ab75327cc57b:02a3cd3621c8dfe5
trace: oid 006c91bb737ce00d:413353bf24fb2349
trace: oid 006ef62f07de0555:0ba7cb4135faa154
trace: oid 001d5bda018eccc9:5df18f575e797d4d
trace: oid 0039a5bc7bd9f19a:02b1e47d07c47a6b
trace: oid 009d55fc1578f224:43f1a46650747015
trace: oid 001644d70ef4ebcd:67b40ab12e7157b5
trace: oid 00b933c420ba0992:1e163c5b1726ab3f
trace: oid 0036cf0d7a22dee3:38ef8b243c3adf39
trace: oid 009fbef06cddcdc4:613602831ea360c8
trace: oid 00bbd319328f9173:549e021c5c0eb520
trace: oid 001e5e3c3d3a16ea:3a88326d65d92ef3
trace: oid 00140885318f4dc5:6d9da95e2b5803f9
trace: oid 00083fe963807ec0:7bcc740e5eb15e81
trace: oid 00756a8d2b38f292:0d22b637071e84c0
trace: oid 00f2fc245734bb23:1e452fff3e2e9e51
trace: oid 00579a065b9f800e:7a697d8b1f29cb31
trace: oid 007d4dd2126b2e16:3dcd2bfa3ef758f7
trace: oid 00fabf8a558e4084:1b060e173d3920eb
trace: oid 00c8576f10aff93c:23124fde79191dc6
trace: oid 003f4701203d95ce:247121bf4bdc5ff4
trace: oid 00be148e37b074ac:2a8dbe75094786ea
trace: oid 00e9673f2eab3baa:10660bab76337f1b
trace: oid 00dff6ce2ecb9af8:34621d6d2edc6363
trace: oid 006b1b060bd35a8f:4e062e95573790d4
trace: oid 003e88a53134f7e2:162ee9cb52e868d8
trace: oid 00c338673333d9a1:102189c36339482f
trace: oid 00e3d2dd00c387b5:5c5265f6198b8fd6
trace: oid 00011d830ff5ae3f:6567efca062319df
trace: oid 00a622ec1fd0ac74:0f6aa0c924bf3211
dbg: created worker thread 0
dbg: created worker thread 1
trace: [1624421207:419384803] t00: Creating objects...
dbg: created worker thread 2
trace: [1624421207:419406249] t01: Creating objects...
dbg: created worker thread 3
trace: [1624421207:419425148] t02: Creating objects...
trace: threads created
trace: [1624421207:419660041] t03: Creating objects...
trace: [1624421208:461481254] t00: Creation done.
trace: [1624421208:461504233] t00: Opening objects...
trace: [1624421208:469365947] t00: Open done.
trace: [1624421208:469382019] t00: Writing objects...
trace: [1624421220:760821639] t02: Creation done.
trace: [1624421220:760849686] t02: Opening objects...
trace: [1624421220:827574754] t01: Creation done.
trace: [1624421220:827597812] t01: Opening objects...
trace: [1624421220:831422947] t02: Open done.
trace: [1624421220:831437789] t02: Writing objects...
trace: [1624421220:871343485] t03: Creation done.
trace: [1624421220:871371394] t03: Opening objects...
trace: [1624421220:876349708] t01: Open done.
trace: [1624421220:876366195] t01: Writing objects...
trace: [1624421221:901601852] t03: Open done.
trace: [1624421221:901634301] t03: Writing objects...
trace: [1624421378:897804856] t00: Write done.
trace: [1624421378:897875064] t00: Reading objects...
trace: [1624421384:010890317] t00: Read done.
trace: [1624421384:010919249] t00: Deleting objects...
trace: [1624421397:033395574] t02: Write done.
trace: [1624421397:033437468] t02: Reading objects...
trace: [1624421404:118561330] t01: Write done.
trace: [1624421404:118608221] t01: Reading objects...
trace: [1624421405:307777950] t00: Deletion done.
dbg: worker thread done 0
trace: [1624421405:311140733] t03: Write done.
trace: [1624421405:311173653] t03: Reading objects...
trace: [1624421407:509010093] t02: Read done.
trace: [1624421407:509033103] t02: Deleting objects...
trace: [1624421410:542550115] t01: Read done.
trace: [1624421410:542592361] t01: Deleting objects...
trace: [1624421414:686992892] t03: Read done.
trace: [1624421414:687018142] t03: Deleting objects...
trace: [1624421436:579154842] t02: Deletion done.
trace: [1624421437:640893546] t01: Deletion done.
dbg: worker thread done 1
dbg: worker thread done 2
trace: [1624421440:558370397] t03: Deletion done.
dbg: worker thread done 3
info: I/O workload is finished.
info: Total: time=[233:477884976] objs=40 ops=400
info: C: [13:451571362] ([1:030954318] per op)
info: O: [1:030169309] ([0:028862164] per op)
info: D: [29:070082196] ([2:583275015] per op)
info: W: [183:403811818] ([3:562019063] per op), 409600 KiB, 2233 KiB/s
info: R: [10:475327035] ([0:156450639] per op), 409600 KiB, 39101 KiB/s
info: done workload 0
info: ---------------------------------------
[root@ssc-vm-5571 ~]# hctl shutdown
Stopping s3server@0x7200000000000001:0x23 at ssc-vm-5571.colo.seagate.com...
Stopping m0d@0x7200000000000001:0x9 (ios) at ssc-vm-5571.colo.seagate.com...
Stopped m0d@0x7200000000000001:0x9 (ios) at ssc-vm-5571.colo.seagate.com
Stopping m0d@0x7200000000000001:0x20 (confd) at ssc-vm-5571.colo.seagate.com...
Stopping hare-hax at ssc-vm-5571.colo.seagate.com...
Stopped hare-hax at ssc-vm-5571.colo.seagate.com
Making sure that RC leader can be re-elected next time
Stopping hare-consul-agent at ssc-vm-5571.colo.seagate.com...
Stopped hare-consul-agent at ssc-vm-5571.colo.seagate.com
Shutting down RC Leader at ssc-vm-5571.colo.seagate.com...
[root@ssc-vm-5571 ~]#
`

**Comment by vaibhavparatwar** [comment]
@daniarherikurniawan are you still facing this?

**Comment by daniarherikurniawan** [comment]
@vaibhavparatwar, @pavankrishnat  I guess this error is related to this [prerequisite](https://github.com/Seagate/cortx-hare#quick-start) "`3 | cortx-hare and cortx-s3server RPMs are installed | all machines`".

However, now I am blocked by another issue (https://github.com/Seagate/cortx-hare/issues/1665) that prevents me from deploying `hare` RPMs. Please approve the PR that will address that problem. 

I have a question, do we need to install `cortx-s3server` RPMs as well (in order to start hare)? There is no link about cortx-s3server tutorial on the Hare's README.md (https://github.com/Seagate/cortx-hare/blob/main/README.md). 
If cortx-s3server is mandatory, then there must be another section in that readme about "Build Cortx-S3Server RPMs". Let me know if this section is needed, I can create a PR for it. Thx

**Comment by vaibhavparatwar** [comment]
@daniarherikurniawan I think ideal document for hare deployment on single node VM is this one - https://github.com/Seagate/cortx-hare/wiki/Hare-provisioning-for-single-node-deployment - Here all the pre-requisits are mentioned and links to all the required README's / QSGs are mentioned. s3server is mentioned in this document as pre-requisit, but I think to start hare cortx-s3server RPMs are optional .. @mssawant @suvratjoshi @knekrasov can confirm on this.

**Comment by daniarherikurniawan** [comment]
@vaibhavparatwar,  Thanks for the suggestion. I will try that link out.

In my case, I was following the main CORTX  quick start (https://github.com/Seagate/cortx/blob/main/QUICK_START.md) => `3. Build just the block storage layer (motr) from the source` => `ii. Run it across a cluster: ` => then there is a link to `Install HARE` that points to hare's README. 

I am hoping that we can make hare's README as clear as possible to avoid any confusion because this file is linked to the main CORTX quick-start guide, which will be visited by many people.

**Comment by vaibhavparatwar** [comment]
@daniarherikurniawan regarding s3server rpms - if we have non-zero s3 instances mentioned in CDF file, then we need s3server rpms as well.

**Comment by daniarherikurniawan** [comment]
@vaibhavparatwar  That totally makes sense! I believe we should add that information in the Hare Readme. For a newbie like me and all other future CORTX developers, we must know that the S3 is not required by hare unless it is mentioned in the CDF file.

**Comment by daniarherikurniawan** [comment]
> Hi @daniarherikurniawan,
> 
> Looks like there is some problem with m0crate file. I tried same at my end, it ran successfully without any error. Here is my output attached.
> [root_20210623_095741.txt](https://github.com/Seagate/cortx-hare/files/6698924/root_20210623_095741.txt)

@pavankrishnat , could you give me some hints on how to test the m0crate? to make sure that m0crate is performing as expected, is there any simple unit test? Thx!

**Comment by daniarherikurniawan** [comment]
Great news.. :) 
I solved the problem by doing these:
1. Addressing the `facter >= 3.14.8` requirements by changing that into just `facter` at `hare.spec`
2. Debug my `m0crate-io.yaml`, thanks to @pavankrishnat for the attached trace.
3. Restart the cortx modules completely by utilizing `lsmod` and then kill one by one in order.
4. Start the hare by running `hctl bootstrap ...` or `hctl start`. 

Out of these 5, I am not sure which step really matters here. 
I will run this test on 3 other servers and see if I can get a solid result. I close this issue, thx! 
cc: @vaibhavparatwar
